scripts/release/uniprot_update/uniprot_xml2sql_isoform.pl

```diff
@@ -440,40 +441,44 @@
                 next;
             }
             $sdi->inflate();
-            
+
             my $sdd = $sdi->db_id;
             print "Updating master sequence...$sdd\t$ac\n";
-                        
+
             $sdi->Created( @{ $sdi->Created } );
             $sdi->Modified( @{ $sdi->Modified } );
             $sdi->add_attribute_value( 'modified', $instance_edit );
 
             updateinstance($sdi, \%values, $_);
-            
+
             $dupl_flag = 1;
-            
+
             ## master sequence update finished
-            
-            $values{'species'} ||= $sdi->species->[0]; #Inherit species instance (if not already specified) from isoform parent
-            foreach my $is_ac ( sort keys %isoids ) {    #isoforms update                
+
+
+            #Inherit species instance (if not already specified) from isoform parent
+            if (!(grep { defined } @{$values{'species'}})) {
+                $values{'species'}->[0] = $sdi->species->[0];
+            }
+            foreach my $is_ac ( sort keys %isoids ) {    #isoforms update

```

Comment:
What does the variable $is_ac represent?

Reply:
It represents the isoform UniProt accession that are keys in the hash.  Variables like these are legacy and have been in the script for many years.  I agree they're cryptic.  Since we're porting to Java, I don't want to spend too much time cleaning up the Perl script, but I'm going to try to make the Java as self-evident as possible.